### PR TITLE
[VyOS]  interfaces resource module updated with new states

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>

##### SUMMARY

partly resolved: https://github.com/ansible-collections/vyos.vyos/issues/24

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

vyos_interfaces
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: tests/integration/targets/vyos_interfaces/tests/cli/parsed.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        running_config: "{{ lookup('file', '_parsed_config.cfg') }}"
+        state: parsed
+    - debug:
+        msg: "{{ ansible_facts['network_resources']['interfaces'] }}"
+    - name: Assert that correct parsing done
+      assert:
+        that: "{{ ansible_facts['network_resources']['interfaces'] | symmetric_difference(result['parsed'])\
+          \ |length == 0 }}"
+
+    - name: Gather the existing running configuration (IDEMPOTENT)
```
Comment (justjais) [inline tests/integration/targets/vyos_interfaces/tests/cli/parsed.yaml:32]: for parsed idempotent tests are not required

File: tests/integration/targets/vyos_interfaces/tests/cli/gathered.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        config:
+        state: gathered
+
+    - name: Assert that gathered dicts was correctly generated
+      assert:
+        that:
+          - "{{ populate | symmetric_difference(result['gathered']) |length == 0\
+            \ }}"
+
+    - name: Gather the existing running configuration (IDEMPOTENT)
```
Comment (justjais) [inline tests/integration/targets/vyos_interfaces/tests/cli/gathered.yaml:24]: for gathered idempotent tests are not required

File: tests/integration/targets/vyos_interfaces/tests/cli/rendered.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            enabled: false
+        state: rendered
+
+    - name: Assert that correct set of commands were generated
+      assert:
+        that:
+          - "{{ rendered['commands'] | symmetric_difference(result['rendered'])\
+            \ |length == 0 }}"
+
+    - name: Structure provided configuration into device specific commands (IDEMPOTENT)
```
Comment (justjais) [inline tests/integration/targets/vyos_interfaces/tests/cli/rendered.yaml:46]: for rendered idempotent tests are not required

File: tests/integration/targets/vyos_interfaces/tests/cli/gathered.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    msg: START vyos_interfaces gathered integration tests on connection={{ ansible_connection
+      }}
+
+- include_tasks: _remove_config.yaml
+
+- include_tasks: _populate.yaml
+
+- block:
+
+    - name: Merge the provided configuration with the exisiting running configuration
```
Comment (justjais) [inline tests/integration/targets/vyos_interfaces/tests/cli/gathered.yaml:12]:
```suggestion
    - name: Gather the provided configuration with the exisiting running configuration
```

File: tests/integration/targets/vyos_interfaces/tests/cli/parsed.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Provide the running configuration for parsing (config to be parsed)
+      register: result
+      vyos.vyos.vyos_interfaces:
+        running_config: "{{ lookup('file', '_parsed_config.cfg') }}"
+        state: parsed
+    - debug:
+        msg: "{{ ansible_facts['network_resources']['interfaces'] }}"
+    - name: Assert that correct parsing done
+      assert:
+        that: "{{ ansible_facts['network_resources']['interfaces'] | symmetric_difference(result['parsed'])\
```
Comment (NilashishC) [inline tests/integration/targets/vyos_interfaces/tests/cli/parsed.yaml:29]: Same comment as https://github.com/ansible-collections/vyos.vyos/pull/26

**Pull request merged**